packages/core/src/domain/configuration/transportConfiguration.ts

```diff
@@ -24,62 +24,80 @@ export function computeTransportConfiguration(
   initConfiguration: InitConfiguration,
   buildEnv: BuildEnv
 ): TransportConfiguration {
-  const endpointBuilders = {
-    logsEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'logs'),
-    rumEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'rum'),
-    sessionReplayEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'sessionReplay'),
-  }
-  const intakeEndpoints: string[] = objectValues(endpointBuilders).map((builder) => builder.buildIntakeUrl())
+  const endpointBuilders = computeEndpointBuilders(initConfiguration, buildEnv)
+  const intakeEndpoints = objectValues(endpointBuilders)
+    .filter(notEmpty)
```

Comment:
why do we need to check for null here?
could we even not do this check by not providing undefined value?

Reply:
It's a tradeoff I made, to have a clearer `computeEndpointBuilders` but let's do it differently then.